src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/TensorPrimitives.netcore.cs

```diff
@@ -3098,6 +3326,802 @@ public static Vector128<float> Invoke(Vector128<float> x, Vector128<float> y)
+                    {
+                        curMax = result[i];
+                        curIn = maxIndex[i];
+                    }
+                    else if (result[i] > curMax)
+                    {
+                        curMax = result[i];
+                        curIn = maxIndex[i];
+                    }
+                }
+                return curIn;
+            }
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static void Invoke(ref Vector512<float> max, Vector512<float> current, ref Vector512<int> maxIndex, Vector512<int> curIndex)
+            {
+                maxIndex = Vector512.ConditionalSelect(Vector512.Equals(max, current).AsInt32(),
+                                    Vector512.ConditionalSelect(IsNegative(max).AsInt32(), curIndex, maxIndex),
+                                    Vector512.ConditionalSelect(Vector512.GreaterThan(max, current).AsInt32(), maxIndex, curIndex));
+
+                max = Vector512.ConditionalSelect(Vector512.Equals(max, current),
+                        Vector512.ConditionalSelect(IsNegative(max), current, max),
+                        Vector512.Max(max, current));
+            }
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static void Invoke(ref Vector512<float> max, Vector512<float> current, ref Vector512<int> maxIndex, Vector512<int> curIndex, Vector512<float> remainderMask)
+            {
+                maxIndex = Vector512.ConditionalSelect(
+                    Vector512.Equals(remainderMask.AsInt32(), Vector512<int>.Zero),
+                    maxIndex,
+                    Vector512.ConditionalSelect(Vector512.Equals(max, current).AsInt32(),
+                    Vector512.ConditionalSelect(IsNegative(max).AsInt32(), curIndex, maxIndex),
+                    Vector512.ConditionalSelect(Vector512.GreaterThan(max, current).AsInt32(), maxIndex, curIndex)));
+
+                max = Vector512.ConditionalSelect(
+                    Vector512.Equals(remainderMask, Vector512<float>.Zero),
+                    max,
+                    MaxOperator.Invoke(max, current));
+            }
+#endif
+        }
+
+        private readonly struct IndexOfMaxMagnitudeOperator : IIndexOfOperator
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static int Invoke(ReadOnlySpan<float> result)
+            {
+                float curMax = result[0];
+                int curIn = 0;
+                if (float.IsNaN(curMax))
+                {
+                    return curIn;
+                }
+
+                for (int i = 1; i < result.Length; i++)
+                {
+                    float current = result[i];
+                    if (float.IsNaN(current))
+                    {
+                        return i;
+                    }
+
+                    if (MathF.Abs(curMax) == MathF.Abs(current))
```

Comment:
Was there a reason you didn't capture the value of Abs in a local like was done before?  https://github.com/dotnet/runtime/pull/93469/files#diff-c9855f90de28628f094b952b640408bbbd586d0d9ac0faf2542b5ba1e289b01fL381
I had the same comment for all instances of these.

Reply:
Nope, no reason. The other places are now using different code and don't have this behavior. Would it be better to capture it in a local instead of doing it twice?